src/routes/utils.ts

```diff
@@ -9,21 +9,46 @@ import { withSiteBaseURL } from '@src/lib/withBaseURL'
 
 const isTestEnv = window?.localStorage.getItem(IS_PLAYWRIGHT_KEY) === 'true'
 
-export const APP_VERSION =
-  isTestEnv && env().NODE_ENV === 'development'
-    ? '0.0.0'
-    : isDesktop()
-      ? // @ts-ignore
-        window.electron.packageJson.version
-      : 'main'
+export function getAppVersion({
+  isTestEnvironment,
+  NODE_ENV,
+  isDesktop,
+}: {
+  isTestEnvironment: boolean
+  NODE_ENV: string | undefined
+  isDesktop: boolean
+}) {
+  if (isTestEnvironment && NODE_ENV === 'development') {
+    return '0.0.0'
+  }
+
+  if (isDesktop) {
+    // @ts-ignore
+    return window.electron.packageJson.version
+  }
+
+  // Web based runtimes
+  if (NODE_ENV === 'development') {
+    return 'dev'
+  }
```

Comment:
After some investigation, I have discovered this won't work after all. Vercel seems to force `NODE_ENV=production` so we'll have to rely on another mechanism to detect the non-production web app.

Reply:
This will work for localhost development which helps for the `DEBUG` check part. Thank you for the clarification on deployed web applications.